edb/schema/delta.py

```diff
@@ -201,18 +202,22 @@ def can_delete(name: sn.Name) -> bool:
                 or not can_delete(y_name)
                 or x_name in renames_x
             ):
+                if (
+                    (x_alter_variants[x] > 1 or can_create(x_name))
+                    and parent_confidence != 1.0
+                ):
+                    confidence = s
+                else:
+                    confidence = 1.0
```

Comment:
Why does parent confidence of 1.0 imply child confidence of 1.0? Because any child change that doesn't affect the parent confidence should be considered to be perfectly confident?

Reply:
Yeah, this is intended to handle the innards of fully-confident `CREATE`.  It's less clear for `ALTER`, because we may be confident about the `ALTER` (via guidance ban on `DROP`/`CREATE`), but that does not necessarily mean full confidence of the child alters.  Will likely need to do more tinkering here, I left a TODO here so we don't forget.